jsapp/js/components/modalForms/connectProjects.es6

```diff
@@ -186,163 +298,293 @@ class ConnectProjects extends React.Component {
+              checked={this.state.isSharingSomeQuestions}
+              onChange={this.onSharingCheckboxChange}
+              label={t('Select specific questions to share')}
             />
-            <label className='connect-projects__label select-errors'>
-              {this.state.fieldsErrors.emptyParent || this.state.fieldsErrors.parent}
-            </label>
           </div>
 
+          {this.state.isSharingSomeQuestions &&
+            <div className='connect-projects__export--multicheckbox'>
+              <span>
+                {t('Select any questions you want to share in the right side table')}
+                {this.state.isLoading &&
+                  renderLoading(t('Updating shared questions'))
+                }
+              </span>
+
+              <MultiCheckbox
+                items={this.state.columnsToDisplay}
+                disabled={this.state.isLoading}
+                onChange={this.onColumnSelected}
+              />
+            </div>
+          }
+        </div>
+      );
+    } else {
+      return (
+        <div className='connect-projects__export'>
+          <div className='connect-projects__export--switch'>
+            <ToggleSwitch
+              onChange={this.onToggleSharingData.bind(this)}
+              label={t('Data sharing disabled')}
+              checked={this.state.isShared}
+            />
+          </div>
+        </div>
+      );
+    }
+  }
+
+  renderImports() {
+    return (
+      <div className='connect-projects__import'>
+        <div className='connect-projects__import--form'>
+          {this.renderSelect()}
+
           <TextBox
             placeholder={t('Give a unique name to the import')}
             value={this.state.newFilename}
             onChange={this.onFilenameChange}
-            errors={this.state.fieldsErrors.emptyFilename ||
-                    this.state.fieldsErrors.filename}
+            errors={this.state.fieldsErrors.filename}
           />
 
-          <bem.KoboButton m='blue'>
+          <bem.KoboButton
+            m='blue'
+            onClick={this.onConfirmAttachment}
+          >
             {t('Import')}
           </bem.KoboButton>
         </div>
-      );
-    } else {
-      return null;
-    }
+
+        {/* Display attached projects */}
+        <ul className='connect-projects__import--list'>
+          <label>{t('Imported')}</label>
+
+          {(!this.state.isInitialised || this.state.isLoading) &&
+            <div className='connect-projects__import--list-item'>
+              {renderLoading(t('Loading imported projects'))}
+            </div>
+          }
+
+          {!this.state.isLoading && this.state.attachedParents.length == 0 &&
+            <li className='connect-projects__no-imports'>
+              {t('No data imported')}
+            </li>
+          }
+
+          {!this.state.isLoading && this.state.attachedParents.length > 0 &&
+            this.state.attachedParents.map((item, n) => {
+              return (
```

Comment:
More personal preference, but maybe this?
```javascript
...
const {filename, parentName, attachmentUrl, etc., etc., etc.} = item
return (
...
)
```

Reply:
@joshuaberetta sorry if I'm missing something, but wouldn't I still have to iterate through `this.state.attachedParents` with some `item` (current value) to get its `filename, parentName, attachmentUrl, etc., etc., etc.`?